src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2345,7 +2395,128 @@ public static void copyImplementation(Design src, Design dest, boolean lockPlace
+			put("FMUX", Arrays.asList("F5LUT", "F6LUT"));
+			put("G_O",  Arrays.asList("G5LUT", "G6LUT"));
+			put("GMUX", Arrays.asList("G5LUT", "G6LUT"));
+			put("H_O",  Arrays.asList("H5LUT", "H6LUT"));
+			put("HMUX", Arrays.asList("H5LUT", "H6LUT"));
+		}};
+
+		// Map from a node to its driver PIP
+		// Note: Some PIPs are bidirectional. But, every PIP performs only one direction.
+		// The direction of a bidirectional PIP is determined from the context, ie., its connecting directional PIPs.
+		// To determine that context, go through directional PIPs first. This process does not support consecutive bidirectional PIPs.
+		Map<Net,Map<Node,PIP>> netToUphillPIPMap = new HashMap<>();
+		// netToPIPs to store PIPs extracted for static nets.
+		Map<Net,Set<PIP>> netToPIPs = new HashMap<>();
+		for (Net net : staticNets) {
+			netToPIPs.put(net, new HashSet<>());
+			Map<Node,PIP> nodeToDriverPIP = new HashMap<>();
+			List<PIP> biPIPs = new ArrayList<>();
+			// TODO: This goes through the whole design. Try to limit it to the target cell, if possible.
+			for (PIP pip : net.getPIPs()) {
+				if (pip.isBidirectional()) {
+					biPIPs.add(pip);
+				} else {
+					nodeToDriverPIP.put(pip.getEndNode(), pip);
+				}
+			}
+			for (PIP pip : biPIPs) {
+				Node stNode = pip.getStartNode();
+				if (nodeToDriverPIP.containsKey(stNode)) {
+					nodeToDriverPIP.put(pip.getEndNode(), pip);
+				} else {
+					nodeToDriverPIP.put(stNode, pip);
+				}
+			}
+			netToUphillPIPMap.put(net, nodeToDriverPIP);
+		}
+
+		Set<String> prohibitBels = new HashSet<>();
+
+		for (SiteInst siteInst : siteInstsOfCells) {
+			// Go through the set of pins being used on this site instance.
+			for (SitePinInst sitePinInst : siteInst.getSitePinInsts()) {
+				if (sitePinInst.isOutPin())
+					continue;
+
+				Net net = sitePinInst.getNet();
+				Map<Node,PIP> nodeToDriverPIP = netToUphillPIPMap.get(net);
+				if (nodeToDriverPIP != null) {
+					Set<PIP> allPIPs = netToPIPs.get(net);
+					// This SitePinInst connects to a static net. Trace and collect all the PIPs to a source.
+					Node node = sitePinInst.getConnectedNode();
+					SitePin sitePin = node.getSitePin();
+					// Backtrack through routing nodes (no SitePin)
+					while ((sitePin == null) || sitePin.isInput()) {
+						PIP pip = nodeToDriverPIP.get(node);
+						allPIPs.add(pip);
+						if (pip.isBidirectional()) {
+							node = pip.getStartNode().equals(node) ? pip.getEndNode() : pip.getStartNode();
+						} else {
+							node = pip.getStartNode();
+						}
+						if (node.getWireName().contains("VCC_WIRE"))
+							break;
+
+						sitePin = node.getSitePin();
+					}
+					if ((sitePin != null) && !node.getWireName().contains("VCC_WIRE"))  { // GND source
+						String  pinName = sitePin.getPinName();
+						String  siteName = sitePin.getSite().getName();
+						List<String> bels = sitePin2Bels.get(pinName);
+						for (String bel : bels) {
+							prohibitBels.add(siteName + "/" + bel);
+						}
```

Comment:
These BELs need to be prohibited because they are used to supply static sources, right? Let's drop that as a comment for the future.

Reply:
comment added.